### ISSUE TITLE
table name add ` symbols when TDengine 

### ISSUE THREAD

**Issue by CRGao** [dbeaver/dbeaver#22161]
### Description

An anomaly occurs when clicking on a table
![截圖 2023-12-13 10 19 36](https://github.com/dbeaver/dbeaver/assets/8327466/5788b1a7-fa00-4c0b-a9d3-d12a5da0f18b)

Example
![截圖 2023-12-13 10 12 37](https://github.com/dbeaver/dbeaver/assets/8327466/9d31b6d8-d3bf-4f15-9fec-283390532be6)

### DBeaver Version

community 23.2.3.202310221533

### Operating System

macOS 14.0

### Database and driver

Driver name : com.taosdata.jdbc:taos-jdbcdriver:RELEASE 3.2.1

**Comment by LonwoLonwo** [comment]
Hello @CRGao 

But what is the rule in this case? I do not see any special symbols in your table name. Why we need quotes here?

**Comment by CRGao** [comment]
I have confirmed here that if the ` symbol is not used
Table names will not be case sensitive.

will not be able to get the table data , table does not exist

![截圖 2023-12-13 10 12 37](https://github.com/dbeaver/dbeaver/assets/8327466/a337bfcc-9af7-4101-8176-377ecab8efc5)

**Comment by LonwoLonwo** [comment]
Any other database object names require quotes in case special symbols are inside these names.
An example of the table name on your screen does not contain special symbols. For this reason - it is not quoted by DBeaver.
So, what is the rule for quoting for TDengine?

**Comment by CRGao** [comment]
Because I used backticks (`) when creating the table name, it resulted in case sensitivity. To correctly retrieve the data during a search, I need to include the backticks again.

Naming Rules (Organized by ChatGPT) :
The page from the TDengine documentation discusses naming conventions and limitations for databases, tables, columns, and tags in TDengine. Key points include:

Naming Rules: Names can start with a letter or underscore, but not a number, and are case-insensitive. A special escape character "`" allows for more naming options while preserving case sensitivity.

Character Set for Passwords: Specifies allowable characters for passwords, excluding quotes, backslashes, and spaces.

General Limitations: This section covers various limits, such as maximum lengths for database names, table names, column names, and tag names. It also addresses limitations on the number of columns, tags, SQL statement length, and the total number of rows.

Table and Column Naming: Explains the constraints and length limits for table and column names, including rules for names with escape characters.

For detailed information and examples, you can visit the [TDengine Documentation](https://docs.taosdata.com/taos-sql/limit/).

**Comment by LonwoLonwo** [comment]
And I'm still confused.

> Names can start with a letter or underscore, but not a number, and are case-insensitive.

So, names are case-insensitive but with a special escape character - they are case-sensitive.
But how we can determine that this name has escaped? Maybe any system table has this info? 

And what if you create a table with the name "aBc" - will it be converted to "ABC" or "abc"?

**Comment by CRGao** [comment]
If backticks are used, you'll get "aBc"; without them, it's "abc".

**Comment by CRGao** [comment]
When I execute the two sql:
```
create table if not exists `device_property_aBc_d0000000001_e_1` (ts timestamp, param bigint, device_name varchar(50), property_type varchar(50));
table name  => device_property_aBc_d0000000001_e_1

create table if not exists device_property_aBc_d0000000001_e_1 (ts timestamp, param bigint, device_name varchar(50), property_type varchar(50));
table name => device_property_abc_d0000000001_e_1
```

**Comment by LonwoLonwo** [comment]
Ok. Could you please try to change these settings in your connection settings?

![2023-12-21 11_14_13-DBeaver Ultimate 23 3 1 - RoleMembersV](https://github.com/dbeaver/dbeaver/assets/45152336/6642bf1c-5391-4dc0-ba8c-6f1a6769d871)

Then reconnect and do some operations with your table.

**Comment by CRGao** [comment]
I have tried adjusting these settings, but the error still persists.

```sql
These are the SQL commands I used to create my table

#step1:
CREATE STABLE `model_custom_property_27NbfMsBLLa_e_1` (`ts` TIMESTAMP, `param` BIGINT) TAGS (`device_name` VARCHAR(50), `property_type` VARCHAR(50));

#step2:
CREATE TABLE `device_property_27NbfMsBLLa_d0000000001_e_1` USING `model_custom_property_27NbfMsBLLa_e_1` (`device_name`, `property_type`) TAGS ("d0000000001", "int");

```

**Comment by LonwoLonwo** [comment]
Also, could you please specify this setting with quotes parameters: \`:\`

![2024-01-25 19_20_36-DBeaver Ultimate 23 3 3 - _postgres 6 1_ Script-122](https://github.com/dbeaver/dbeaver/assets/45152336/36ba5f5f-3213-416e-bbd6-47c4a7d4acea)

**Comment by LonwoLonwo** [comment]
Fixed. The fix will be available in the next DBeaver version, 23.3.4, on 05-02-2024.
